packages/components/src/list/test/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, getAllByRole, render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import List from '../index';
+
+describe( 'List', () => {
+	it( 'should have aria roles for items', () => {
+		const clickHandler = jest.fn();
+		const listItems = [
+			{
+				title: 'WooCommerce.com',
+				href: 'https://woocommerce.com',
+			},
+			{
+				title: 'Click me!',
+				onClick: clickHandler,
+			},
+		];
+
+		const { container } = render( <List items={ listItems } /> );
+
+		expect( getAllByRole( container, 'menuitem' ) ).toHaveLength( 2 );
+	} );
+
+	it( 'should support `onClick` for items', () => {
+		const clickHandler = jest.fn();
+		const listItems = [
+			{
+				title: 'WooCommerce.com',
+				href: 'https://woocommerce.com',
+			},
+			{
+				title: 'Click me!',
+				onClick: clickHandler,
+			},
+		];
+
+		const { container } = render( <List items={ listItems } /> );
+
+		fireEvent.click( getAllByRole( container, 'menuitem' )[ 1 ] );
```

Comment:
`getByRole` has some nice querying features if you pass in a second argument.

```js
fireEvent.click( screen.getByRole(  'menuitem', { name: 'WooCommerce.com' } ) );
```

Reply:
Agreed. Updated!